Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -22,6 +28,171 @@ def __init__(self, port):
         super().__init__(self.message)
 
 
+class InvalidSingleSelectException(Exception):
```

Comment:
These exceptions, and the exception below for invalid port value, are effectively the same. Just make one generic exception for all invalid parameter values.

Reply:
Generally, custom exceptions should provide a default message, but have a message parameter that can be overridden:

```
class InvalidParameterValue(Exception):
    def __init__(self, value, parameter_name, valid_arguments=[], message=""):
        if not message:
            ...
```

This allows for flexibility if a similar error is captured but has slightly different criteria. It's also generally recommended to avoid ending your exception class names with `Exception` - either omit it or use `Error`/`Warning` as appropriate.

Reply:
Okay, I'll create a generic exception for the same & do the changes for message as well.